public/src/components/teacher/CourseGroup.tsx

```diff
@@ -192,10 +192,11 @@ export class CourseGroup extends React.Component<ICourseGroupProps, ICourseGroup
     private async handleActionOnClick(group: Group, link: ILink): Promise<void> {
         switch (link.uri) {
             case "approve":
-                group.setStatus(Group.GroupStatus.APPROVED);
                 const ans = await this.props.courseMan.updateGroup(group);
                 if (ans.getCode() !== 0) {
-                this.generateErrorMessage(ans);
+                    this.generateErrorMessage(ans);
```

Comment:
I see this pattern a lot:
```typescript
if (ans.getCode() !== 0) {
    this.generateErrorMessage(ans);
} else {
   // do something different each time the pattern is used...
}
```
Would it make sense to define a closure/function for this, to which you pass `ans` and the code in the else-clause?

Reply:
Yeah, that sounds like a good idea.